src/plugins/discover/public/application/components/discover.tsx

```diff
@@ -203,6 +206,14 @@ export function Discover({
     [opts, state]
   );
 
+  const onEditRuntimeField = () => {
+    if (refreshAppState) {
+      refreshAppState();
+    }
+    // needed to trigger refresh of table cells
+    setShouldRerender(!shouldRerender);
```

Comment:
Triggering a re-render won't work when you edited  runtime field's code.
Because this is triggered before the results were received from ES. Which means, once they are here, there is no update done. Old value stays in the table. You could use `fetch` directly without using `refetch$`. 

```
await fetch();
setShouldRerender(!shouldRerender);
```